plugins/com.google.cloud.tools.eclipse.appengine.facets/src/com/google/cloud/tools/eclipse/appengine/facets/WebProjectUtil.java

```diff
@@ -56,12 +63,18 @@ public static IFolder getWebInfDirectory(IProject project) {
       }
     }
     // Otherwise it's seemingly fair game
-    IFolder defaultLocation = project.getFolder(DEFAULT_WEB_PATH).getFolder(WEB_INF);
-    if (defaultLocation.exists()) {
-      return defaultLocation;
-    } else {
-      return null;
+    for (String possibleWebInfContainer : DEFAULT_WEB_PATHS) {
+      // simplify mocking: get the location as two parts and check for null despite that getFolder()
+      // should be @NonNull
+      IFolder defaultLocation = project.getFolder(possibleWebInfContainer);
+      if (defaultLocation != null && defaultLocation.exists()) {
+        defaultLocation = defaultLocation.getFolder(WEB_INF);
+        if (defaultLocation != null && defaultLocation.exists()) {
+          return defaultLocation;
+        }
+      }
```

Comment:
Hmm, kinda: the detectors uses WebProjectUtil#findInWebInf() to find the `appengine-web.xml` file, which uses these lookup locations.  The AppEngine Java SDK demos have the `appengine-web.xml` in `war/`, not `src/main/webapp`.